PatientWebApplication/HealthClinic.DAL/Service/EPrescriptionService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using HealthClinic.CL.Adapters;
+using HealthClinic.CL.DbContextModel;
+using HealthClinic.CL.Dtos;
+using HealthClinic.CL.Model.Pharmacy;
+using HealthClinic.CL.Repository;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace HealthClinic.CL.Service
+{
+    public class EPrescriptionService
+    {
+        public EPrescriptionRepository EPrescriptionRepository { get; }
+        public IEPrescriptionRepository IPrescriptionRepository { get; set; }
+        public EPrescriptionService() { }
+
+        public EPrescriptionService(MyDbContext context)
+        {
+            EPrescriptionRepository = new EPrescriptionRepository(context);
+        }
+        public EPrescriptionService(IEPrescriptionRepository prescriptionRepositoy)
+        {
+            IPrescriptionRepository = prescriptionRepositoy;
+        }
+        public List<EPrescription> GetAll()
+        {
+            return EPrescriptionRepository.GetAll();
+        }
+        public List<EPrescription> GetAllForStub()
+        {
+            return IPrescriptionRepository.GetAll();
+        }
+        public EPrescription Create(EPrescriptionDto dto)
+        {
+            EPrescription prescription = EPrescriptionAdapter.EPrescriptionDtoToEPresctiption(dto);
+            return EPrescriptionRepository.Create(prescription);
+        }
+
+        public String GeneratePrescriptionForPatient(string medicalIdNumber)
+        {
+            List<EPrescription> prescriptions = GetAllForStub();
+            foreach (EPrescription prescription in prescriptions)
+            {
+                if (prescription.MedicalIDNumber.Equals(medicalIdNumber))
+                {
+                    return getTextForPrescription(prescription);
+                }
+            }
+            return "";
```

Comment:
Write this in form:
"EPrescription prescription = GetAllForStub().SingleOrDefault(prescriptionIt => (prescriptionIt.MedicalIDNumber.Equals(medicalIdNumber)));
return (prescritption!=null?getTextForPrescription(prescription):"");"